h/queue.py

```diff
@@ -48,6 +45,27 @@ def get_writer(settings):
     return nsqd
 
 
+def resolve_topic(topic, namespace=None, settings=None):
+    """
+    Return a resolved name for the requested topic.
+
+    This uses the passed `namespace` to resolve the topic name, or,
+    alternatively, a pyramid settings object.
+    """
+    if namespace is not None and settings is not None:
+        raise ValueError('you must provide only one of namespace or settings')
+
+    if settings is not None:
+        ns = settings.get('nsq.namespace')
```

Comment:
Oh, I think I see what you're asking -- no, not really, because the 'nsq.namespace' path is an implementation detail of this module. Asking `h.streamer` to pull it out of the application settings object by name is violating that encapsulation. That's the whole point of this function -- if I wasn't concerned with encapsulating the logic associated with namespacing within this module I could just pull `nsq.namespace` out of the settings object in `h.streamer` and prepend it to the topic names.